tests/streamtests/streamtests.cc

```diff
@@ -0,0 +1,489 @@
+// Copyright 2021 Paolo Jovon, All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <cstring>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <memory>
+#include <utility>
+#include <vector>
+#include <iostream>
+#include <fstream>
+#include "gl_format.h"
+#include "ktx.h"
+#include "gtest/gtest.h"
+
+namespace
+{
+
+constexpr const char SAMPLE_KTX1[] = "pattern_02_bc2.ktx";
+constexpr const char SAMPLE_KTX2[] = "pattern_02_bc2.ktx2";
+
+std::string testImagesPath;
+
+// For AppleClang 12, since it can't find `std::make_unique<>`
+template <typename T, typename... Args>
+inline std::unique_ptr<T> make_unique(Args&&... args)
```

Comment:
Shouldn't this be ifdefed for AppleClang? Or is it okay if all platforms use this?

Reply:
I think it should work on every platform. I'm not too sure of why AppleClang isn't picking it up from `<memory>`, though.

Reply:
The 3 make_unique definitions are inside an `#if _LIBCPP_STD_VER > 11`. I don't know if you've done anything different for building this test but all other c++ in the project builds with c++11.

Reply:
I will force the `CXX_STANDARD` to C++11 for this particular executable. I think appleclang still defaults to C++98 (ew), since I also had problems with the Rust build for the same reason.